crates/matrix-sdk-base/src/rooms/members.rs

```diff
@@ -49,6 +52,33 @@ pub struct RoomMember {
 }
 
 impl RoomMember {
+    pub(crate) fn from_parts(member_info: MemberInfo, room_info: &MemberRoomInfo<'_>) -> Self {
+        let MemberInfo { event, profile, presence } = member_info;
+        let MemberRoomInfo {
+            power_levels,
+            max_power_level,
+            room_creator,
+            users_display_names,
+            ignored_users,
+        } = room_info;
+
+        let is_room_creator = room_creator.as_ref().map_or(false, |u| u == event.user_id());
```

Comment:
```suggestion
        let is_room_creator = room_creator.as_ref().is_some_and(|u| u == event.user_id());
```

Reply:
(and two times more below)

Reply:
Or maybe

```suggestion
        let is_room_creator = room_creator.as_deref() == Some(event.user_id());
```

if that works?